desktop/ui/src/main/java/org/datacleaner/windows/HadoopConnectionToNamenodeDialog.java

```diff
@@ -0,0 +1,282 @@
+        _saveButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+
+                URI nameNodeUri;
+                try {
+                    nameNodeUri = new URI("hdfs", null, _hostTextField.getText(), Integer.parseInt(_portTextField
+                            .getText()), "/", null, null);
+                    DirectConnectionHadoopClusterInformation newServer = new DirectConnectionHadoopClusterInformation(
+                            _nameTextField.getText(), _descriptionTextField.getText(), nameNodeUri);
+                    _savedServer = newServer;
+                    if (_directConnection != null) {
+                        _mutableServerInformationCatalog.removeServer(_directConnection);
+                    }
+                    _mutableServerInformationCatalog.addServerInformation(newServer);
+                    dispose();
+                } catch (URISyntaxException e1) {
+                    setStatusError(e1);
+                    setSaveButtonEnabled(false);
+                } catch (Exception exception) {
+                    setStatusError(exception);
+                    setSaveButtonEnabled(false);
+                }
+
+            }
+        });
+
+        _cancelButton = WidgetFactory.createDefaultButton("Cancel", IconUtils.ACTION_CANCEL);
+        _cancelButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                HadoopConnectionToNamenodeDialog.this.close();
+            }
+        });
+
+        if (directConnection != null) {
+            _nameTextField.setText(directConnection.getName());
+            _nameTextField.setEnabled(false);
+            _hostTextField.setText(directConnection.getNameNodeUri().getHost());
+            _portTextField.setText("" + directConnection.getNameNodeUri().getPort());
+            final String description = directConnection.getDescription();
+            if (description != null) {
+                _descriptionTextField.setText(description);
+            }
+        }
+
+        _nameTextField.getDocument().addDocumentListener(new DCDocumentListener() {
+            @Override
+            protected void onChange(DocumentEvent event) {
+                validateAndUpdate();
+            }
+        });
+
+        _hostTextField.getDocument().addDocumentListener(new DCDocumentListener() {
+            @Override
+            protected void onChange(DocumentEvent event) {
+                validateAndUpdate();
+            }
+        });
+
+        _portTextField.getDocument().addDocumentListener(new DCDocumentListener() {
+            @Override
+            protected void onChange(DocumentEvent event) {
+                validateAndUpdate();
+            }
+        });
+
+        _descriptionTextField.getDocument().addDocumentListener(new DCDocumentListener() {
```

Comment:
You should probably reuse a singe DCDocumentListener instance instead of creating multiple ones.